### ISSUE TITLE
Failed to get required imports for function

### ISSUE THREAD

**Issue by mbercx** [aiidateam/aiida-workgraph#175]
While trying the nice new `pythonjob` decorator:

```python
from aiida_workgraph import task
from functions import generate_structures

decorated_function = task.pythonjob()(generate_structures)
```

No error is raised, but the following message is printed:

```
Failed to get required imports for function generate_structures: type object 'list' has no attribute '_name'
```

The `generate_structures` function is the following:

```python
def generate_structures(
        structure: Atoms,
        strain_lst: list
    ) -> list[Atoms]:

    structure_lst = []

    for strain in strain_lst:
        structure_strain = structure.copy()
        structure_strain.set_cell(
            structure_strain.cell * strain**(1/3), 
            scale_atoms=True
        )
        structure_lst.append(structure_strain)

    return structure_lst
```

I'm assuming the issue is the returned `list`? This issue also made me wonder how to deal with `list` outputs. The user here would probably want to have a list of linked `AtomsData` nodes, but I suppose that is not what will happen. Off the top of my head, I'm considering if the decorator function should have some kind of input argument that allows the user to specify that the list should be expanded as several linked output nodes.

**Comment by superstar54** [comment]
> Failed to get required imports for function generate_structures: type object 'list' has no attribute '_name'

There is an error when the WorkGraph tries to inspect the type annotations, and it fails to inspect the `list`
```
strain_lst: list
```
This should be a bug. I will try to fix it this morning.  For the moment, you can remove the type annotations, and try.


There are three possible solutions for the list outputs.

- Use the function as it is so that PythonJob will Pickle the whole list and return a `GeneralData` (`PickileData`); you then pass this data into the next task. Note that the data provenance is lost.
- Change the list to a dict, and set the `output` as `Namespace`, this will serialize the value of the dict. Thus each Atoms will be an AtomsData, and can used by the next task. The data provenance is kept.


```python
def generate_structures(
        structure: Atoms,
        strain_lst: list
    ) -> list[Atoms]:

    structure_dict = {}

    for strain in strain_lst:
        structure_strain = structure.copy()
        structure_strain.set_cell(
            structure_strain.cell * strain**(1/3), 
            scale_atoms=True
        )
        structure_dict[f"strain_{strain}"] = structure_strain

    return structure_dict

decorated_function = task.pythonjob(outputs=[{"name": "structure_dict", "identifier": "Namespace"}])(generate_structures)
```
- Support the list output, but convert to a dict by us with a default label index, and convert it back to a list when using it. **This is not support yet**, but we can add this.

What do you think?

**Comment by superstar54** [comment]
Hi @mbercx , the `get required imports` bug is fixed in #177 .

However, the feature that allows users to return a list of AiiDA nodes is not yet. I opened a new issue #189 .
